common/scripts/poll-npm-package-published.mjs

```diff
@@ -44,7 +44,9 @@ const main = async () => {
       process.exitCode = 0;
       return;
     } else if ((new Date() - startTime) > TIMEOUT_TIME_MS) {
-      throw new Error('Failed to find package on the npm registry');
+      console.error("Failed to find package on the npm registry");
```

Comment:
I recommend adding a comment that you had to set exit code explicit for the GH action to detect failure.

Though... I'm curious why that is the case.

Reply:
If you're taking requests, I'd rather launch the script with `--unhandled-rejections=strict` ;)

![image](https://user-images.githubusercontent.com/82062616/183454363-5cd67435-2038-46c6-8dce-e77528980877.png)

Reply:
We could add a flag, but seems for backwards compat they don't have a non-zero error code when it fails:
![image](https://user-images.githubusercontent.com/2684369/183454522-10700ff8-c2c1-41c7-bff7-7a48786f79aa.png)

Perhaps doing
```
process.existCode = 1;
throw new Error(...)
```
would work also?

Reply:
`--unhandled-rejections=strict` seems to have the intended behavior:

For,
```typescript
async function badUn() {
    throw new Error('do not handle me');
}

badUn();
```

```
@prprabhu-ms ➜ /workspaces/communication-ui-library/packages/react-composites (prprabhu/deflake-config-page ✗) $ node --unhandled-rejections=strict scripts/why-wont-you-die.mjs; echo $?
file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:2
    throw new Error('do not handle me');
          ^

Error: do not handle me
    at badUn (file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:2:11)
    at file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:5:1
    at ModuleJob.run (internal/modules/esm/module_job.js:170:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
1
```

But more importantly, all this is only needed if you don't `await` the async function. If you modify that toy example to:

```typescript
async function badUn() {
    throw new Error('do not handle me');
}

await badUn();
```

Then you get

```
@prprabhu-ms ➜ /workspaces/communication-ui-library/packages/react-composites (prprabhu/deflake-config-page ✗) $ node scripts/why-wont-you-die.mjs; echo $?
file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:2
    throw new Error('do not handle me');
          ^

Error: do not handle me
    at badUn (file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:2:11)
    at file:///workspaces/communication-ui-library/packages/react-composites/scripts/why-wont-you-die.mjs:5:7
    at ModuleJob.run (internal/modules/esm/module_job.js:170:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
1
```

Without any extra flags.

Reply:
So you simply need to `await main()` on the last line.

Reply:
Updated !